packages/polymorphic/src/Polymorphic.hooks.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import { PolymorphicAs, PolymorphicRef } from './Polymorphic.types';
+
+/**
+ * A wrapper around `React.useRef`
+ * that provides an accurately typed Ref object
+ */
+export const usePolymorphicRef = <E extends PolymorphicAs>(_?: E) => {
+  // By accepting a prop with a generic type, we can type this function using JS-domain vars
```

Comment:
What is JS-domain vars?

Reply:
This refers to real JavaScript variables (here the `_` arg) as opposed to Typescript generic variables (here the type `E`)

I'll update the comment to clarify, but essentially, by accepting an argument of the same generic type, we can set the type `E` using a JS variable.

i.e. instead of calling `usePolymorphicRef<'div'>()` we can instead call `usePolymorphicRef('div')`